src/from-promise.ts

```diff
@@ -34,14 +34,14 @@ function caseImpl<U, T>(handlers: {
     pending?: (t?: T) => U
     fulfilled?: (t: T) => U
     rejected?: (e: any) => U
-}): U {
+}, defaultFulfilled: boolean = false): U {
     switch (this.state) {
         case PENDING:
             return handlers.pending && handlers.pending(this.value)
         case REJECTED:
             return handlers.rejected && handlers.rejected(this.value)
         case FULFILLED:
-            return handlers.fulfilled && handlers.fulfilled(this.value)
+            return defaultFulfilled ? this.value : handlers.fulfilled && handlers.fulfilled(this.value)
```

Comment:
```suggestion
            return handlers.fulfilled ? handlers.fulfilled(this.value) :  this.value
```
@JefHellemans What about resolving the `fulfilled` automatically if the fulfilled method did not passed?
This way you will gain two things:
- You wot break the api
- You  will get a cleaner way to get what you wished for.

@mweststrate what do you think about it?

Reply:
It actually is a breaking change, as before you would get `undefined`. I just can't really think of a use case where you don't want to get the `fulfilled` value though, so this default might make more sense. 

After letting this rest for a while, I have to admit that my suggested change doesn't really add a lot of value. @mweststrate, I'd say either give this solution from @ItamarShDev a go or close the PR.

Reply:
Agreed, I think @ItamarShDev proposal is the neatest. (one could wonder whether on a missing reject we should explicitly `throw` instead of doing nothing, which is kinda eating exceptions?). That would definitely be a breaking change though. Although I think that is fine, this package didn't have a major bump in a long time :)

Reply:
I'll implement it tomorrow 🙂